blockstore/splitstore/splitstore.go

```diff
@@ -1126,6 +1184,7 @@ func (s *SplitStore) beginTxnProtect() *sync.WaitGroup {
 }
 
 func (s *SplitStore) beginTxnMarking(markSet MarkSet) {
+	// No lock? Have we tested this with the golang race detector?
```

Comment:
it's only invoked by compaction exactly once, so it is protected effectively by the `compacting` lock.

Reply:
Hm. But it's _read_ elsewhere, right? Have you run tests with the race detector?

Reply:
yes, I have, but we have just one simple unit test so it's not that great of a confidence builder.